docs: jetbrains docs in gitpod yaml ref (#2021)
gitpod-io/website: gitpod/docs/references/gitpod-yml.md

- [`image`](#image)
  - [`image.file`](#imagefile)
  - [`image.context`](#imagecontext)

> - [`jetbrains`](#jetbrains) `🧪`

Reviewer: I appreciate it looks like we've copied the warning / experimental boxes from the intelliJ docs page, however since this language (experimental) is what users will need to set expectations about a feature, we should be a little careful. I'd suggest we remove the emoji (since it's not language or a pattern that we've defined anywhere). I think the warning boxes below we can leave for this PR, however, would be good to do another pass to update (most likely to "beta" rather than "experimental") but we can do that separately: https://github.com/gitpod-io/website/issues/2054

For context, see [internal RFC about "in preview" vs "beta" language use](https://www.notion.so/gitpod/RFC-Beta-vs-In-Preview-vs-aea46a72c1864c38ab54f0777ac8343b). 

🙏

Author: @loujaybee agreed, I've updated the PR removing the emoji 🧪

</div>

## `jetbrains` `🧪`

Define the integration between Gitpod and JetBrains IDEs.

<div class="overflow-x-auto">

| Type     | Default   |
| -------- | --------- |
| `object` | `<empty>` |

</div>

> > **Please note:** This feature is currently experimental and is a subject to change. [Send feedback](https://github.com/gitpod-io/gitpod/issues/6740).

Reviewer 2: https://github.com/gitpod-io/gitpod/issues/6740 is about prebuilds, we probably should drop it for plugins and our point to another way to provide feedback

Author: I am not sure what you mean 🤔

Reviewer 2: docs are about plugins and general jetbrains section, but https://github.com/gitpod-io/gitpod/issues/6740 is only about indexing feedback

Author: ah, I got you now! I didn't realise that comment was linking to a specific issue 🤦‍♂️

I found a good alternative: https://github.com/gitpod-io/gitpod/issues/6576

PR updated

> > **Please note:** This feature is currently experimental and is a subject to change. [Send feedback](https://github.com/gitpod-io/gitpod/issues/6576).

Reviewer: Could we nest these disclaimers under the title directly? 🙏 

I think we can improve the formatting in: https://github.com/gitpod-io/website/issues/2054

> ## `jetbrains.plugins` `🧪`

Reviewer: The navigation bar on the right looks different for JetBrains compared to the other products, e.g. `image`, and `ports` (see screenshot for `>` caret), I guess that's because they're using a nested header? 

![image](https://user-images.githubusercontent.com/5528307/168575432-f9b9c2b0-200f-4015-abea-31c4aa69e01f.png)

Author: Good catch! Updated ✨

<img width="243" alt="Screenshot 2022-05-16 at 14 30 40" src="https://user-images.githubusercontent.com/2318450/168592825-05bbd8d3-8e93-4883-bd09-70828961096a.png">
 
I had to make the child headers `###` instead of `##`

> Define a list of plugins which should be installed for all JetBrains products for users of this workspace. To find the plugin identifier, from the [JetBrains Marketplace](https://plugins.jetbrains.com), find the desired plugin, open the 'Versions' tab, select any version and copy the 'Plugin ID' (like `${publisher}.${name}`).

Reviewer: Should we note here that not all JetBrains plugins are compatible with all IDE's? (@akosyakov do you know what the expected failure mode will be if an incompatible IDE tries to install a plugin?)

![image](https://user-images.githubusercontent.com/5528307/168577634-7b5df1db-db3b-4b3d-989c-c3b6c6927cbd.png)

Reviewer: Actually, maybe we can just mention "compatible" IDE's? 🤔 [[1](https://github.com/gitpod-io/website/pull/2021/files#r873589214)]

Author: let's go ahead with "compatible" for now 👍

> Define a list of plugins which should be installed for all JetBrains products for users of this workspace. To find the plugin identifier, from the [JetBrains Marketplace](https://plugins.jetbrains.com), find the desired plugin, open the 'Versions' tab, select any version and copy the 'Plugin ID' (like `${publisher}.${name}`).

Reviewer: ```suggestion
Define a list of plugins which should be installed for all JetBrains IDEs when starting a workspace. To find the plugin identifier, from the [JetBrains Marketplace](https://plugins.jetbrains.com), find the desired plugin, open the 'Versions' tab, select any version and copy the 'Plugin ID' (like `${publisher}.${name}`).
```

<div class="overflow-x-auto">

| Type    | Default   |
| ------- | --------- |
| `array` | `<empty>` |

</div>

> **Please note:** This feature is currently experimental and is a subject to change. [Send feedback](https://github.com/gitpod-io/gitpod/issues/6576).

## `jetbrains.[product]` `🧪`

Define the integration between Gitpod and a specific JetBrains IDE. Install plugins and configure prebuilds to speed up the IDE indexing.

Specify the 'product' with one of the following values:

- `intellij`
- `goland`
- `pycharm`
- `phpstorm`

<div class="overflow-x-auto">

| Type     | Default   |
| -------- | --------- |
| `object` | `<empty>` |

</div>

> **Please note:** This feature is currently experimental and is a subject to change. [Send feedback](https://github.com/gitpod-io/gitpod/issues/6576).

## `jetbrains.[product].plugins`

> Define a list of plugins which should be installed for a specific JetBrains product for users of this workspace. To find the plugin identifier, from the [JetBrains Marketplace](https://plugins.jetbrains.com), find the desired plugin, open the 'Versions' tab, select any version and copy the 'Plugin ID' (like `${publisher}.${name}`).

Reviewer: ```suggestion
Define a list of plugins which should be installed for a specific JetBrains product for users of the project. To find the plugin identifier, from the [JetBrains Marketplace](https://plugins.jetbrains.com), find the desired plugin, open the 'Versions' tab, select any version and copy the 'Plugin ID' (like `${publisher}.${name}`).
```

Author: Updated 👍

<div class="overflow-x-auto">

| Type    | Default   |
| ------- | --------- |
| `array` | `<empty>` |

</div>

**Example**

```yaml
jetbrains:
  intellij:
    plugins:
      - zielu.gittoolbox
      - izhangzhihao.rainbow.brackets
```

> **Please note:** This feature is currently experimental and is a subject to change. [Send feedback](https://github.com/gitpod-io/gitpod/issues/6576).

> ## `jetbrains.[product].prebuilds` `🧪`

Reviewer: This isn't a fully-formed thought (and I guess my thought is more aimed at @akosyakov) but having `prebuilds` nested under the IDE's seems odd, as it feels more like it should be a feature at the top level, and apply the properties underneath. Something worth discussing for the longer-term...

Define whether Gitpod enables prebuilds for a specific JetBrains IDE.

<div class="overflow-x-auto">

| Type     | Default   |
| -------- | --------- |
| `object` | `<empty>` |

</div>

**Example**

```yaml
jetbrains:
  intellij:
    prebuilds:
      version: stable
```

The `version` is defined as follows:

<div class="overflow-x-auto">

| Type     | Default  | Values                     |
| -------- | -------- | -------------------------- |

> | `string` | `stable` | `stable`, `latest`, `both` |

Reviewer: I don't think we currently define anywhere the difference between these versions (i.e how a user would know the difference between latest and stable). We have [an FAQ to understand which version of VS Code we're running]( https://gitpod.io/docs/ides-and-editors/faqs#what-version-of-vs-code-runs-in-gitpod-in-the-browser) in the browser, it'd be good to have a section which also explains the difference between the IntelliJ versions, and explain when we update them, etc. I'd be happy to create a separate issue for this, and we can do as a follow up 🙏

Author: Agreed, issue created: https://github.com/gitpod-io/website/issues/2057

I am happy to take of it as a follow-up task :)
